ksp_plugin/manœuvre.hpp

```diff
@@ -0,0 +1,85 @@
+﻿#pragma once
+
+#include <optional.hpp>
+
+#include "geometry/named_quantities.hpp"
+#include "physics/discrete_trajectory.hpp"
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+
+using geometry::Instant;
+using geometry::Vector;
+using physics::DiscreteTrajectory;
+using quantities::Force;
+using quantities::Mass;
+using quantities::SpecificImpulse;
+using quantities::Speed;
+using quantities::Time;
+using quantities::Variation;
+
+namespace ksp_plugin {
+
+// This class represents a constant-thrust inertial burn.
+template<typename Frame>
+class Manœuvre {
+ public:
+  Manœuvre(Force const& thrust,
+           Mass const& initial_mass,
+           SpecificImpulse const& specific_impulse,
+           Vector<double, Frame> const& direction);
+  ~Manœuvre() = default;
+
+  Force thrust() const;
```

Comment:
Since the construction parameters cannot be changed and don't have alternative representations, I'd return them by const&.